### ISSUE TITLE
Wrong indentation on C# initialiser inside a function call

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1263]
expected
```csharp
using System.Collections.Generic;

class C
{
    public void S()
    {
        Action localMethod = () => {
                SomeClass.OtherMethod(new Dictionary<string, string>
                {
                    {"a", "one"},
                    {"b", "two"},
                    {"c", "three"}
                });
            };
    }
}
```
actual
```csharp
using System.Collections.Generic;

class C
{
    public void S()
    {
        Action localMethod = () => {
                SomeClass.OtherMethod(new Dictionary<string, string>
            {
                {"a", "one"},
                {"b", "two"},
                {"c", "three"}
            });
            };
    }
}
```

```
m_Mixers.Add(
                    new WeightInfo
                {
                    parentMixer = parent,
                    mixer = node,
                    port = port,
                    modulate = (type == typeof(AnimationLayerMixerPlayable))
                }
                    );
```

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60015-UNI-14131.cs" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60015-UNI-14131.cs"
index 3473452..d49272f 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60015-UNI-14131.cs"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60015-UNI-14131.cs"
@@ -1,26 +1,26 @@
 using System.Collections.Generic;
 
 class C
 {
     public void S()
     {
         Action localMethod = () => {
                 SomeClass.OtherMethod(new Dictionary<string, string>
-                {
-                    {"a", "one"},
-                    {"b", "two"},
-                    {"c", "three"}
-                });
+            {
+                {"a", "one"},
+                {"b", "two"},
+                {"c", "three"}
+            });
             };
 
-            m_Mixers.Add(
-                new WeightInfo
-                {
-                    parentMixer = parent,
-                    mixer = node,
-                    port = port,
-                    modulate = (type == typeof(AnimationLayerMixerPlayable))
-                }
+        m_Mixers.Add(
+            new WeightInfo
+        {
+            parentMixer = parent,
+            mixer = node,
+            port = port,
+            modulate = (type == typeof(AnimationLayerMixerPlayable))
+        }
             );
     }
 }
```

Debug support files:
---
[60015-UNI-14131.cs](https://gist.githubusercontent.com/mihaipopescu/30a71f33bfc297840fad5b526877db92/raw/3473452ec37303ef4f88a844e83cba0fa706a37e/60015-UNI-14131.cs)
[60015-UNI-14131.cs.log](https://gist.githubusercontent.com/mihaipopescu/30a71f33bfc297840fad5b526877db92/raw/8e1536db6dd14454fd3e13a51b0ab3ab327cf810/60015-UNI-14131.cs.log)
[60015-UNI-14131.cs.unc](https://gist.githubusercontent.com/mihaipopescu/30a71f33bfc297840fad5b526877db92/raw/97163b810f144e08b137dac8a02923403270e4c0/60015-UNI-14131.cs.unc)
[Uncrustify.CSharp.cfg](https://gist.githubusercontent.com/mihaipopescu/30a71f33bfc297840fad5b526877db92/raw/1d064c1fabb76ce328dd015ed40023207de20ca1/Uncrustify.CSharp.cfg)
[UNI-14131.cs](https://gist.githubusercontent.com/mihaipopescu/30a71f33bfc297840fad5b526877db92/raw/3473452ec37303ef4f88a844e83cba0fa706a37e/UNI-14131.cs)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by CDanU** [comment]
Shouldn't you expect something like this:
```cpp
using System.Collections.Generic;

class C
{
    public void S()
    {
        Action localMethod = () = > {
            SomeClass.OtherMethod(new Dictionary<string, string>
            {
                {"a", "one"},
                {"b", "two"},
                {"c", "three"}
            });
        }

        m_Mixers.Add(
            new WeightInfo
            {
                parentMixer = parent,
                mixer = node,
                port = port,
                modulate = (type == typeof(AnimationLayerMixerPlayable))
            });
    }
}
```
?

**Comment by GinoGalotti** [comment]
Yes, you're right. Yours is what we're expecting (we messed up the example a little bit). The current solution is still wrongly indenting the last bit.

```
        m_Mixers.Add(
            new WeightInfo
        {
                parentMixer = parent,
                mixer = node,
                port = port,
                modulate = (type == typeof(AnimationLayerMixerPlayable))
        });
```

**Comment by mihaipopescu** [comment]
> not entirely sure what a delegate in this context is.

in the first context, the delegate is after `=>`

**Comment by gmaurel** [comment]
Fixed at 94bdce3a37d0492f7d458681f3aab755d5510bcd, Jan 19 18:54:29 2018
